uefi_test.go

```diff
@@ -43,11 +44,57 @@ func TestAppendUEFIFirmwareDeviceConfig(t *testing.T) {
 		t.Errorf("Expected non-empty qemuParams, found %s", c.qemuParams)
 	}
 
-	expected := "-drive if=pflash,format=raw,readonly,file=OVMF_CODE.fd -drive if=pflash,format=raw,file=OVMF_VARS.fd"
+	expected := "-drive if=pflash,format=raw,readonly=on,file=OVMF_CODE.fd -drive if=pflash,format=raw,file=OVMF_VARS.fd"
 	result := strings.Join(c.qemuParams, " ")
 	if expected != result {
 		t.Fatalf("Failed to append parameters\nexpected[%s]\n!=\nfound   [%s]", expected, result)
 	}
 }
 
+func TestAppendUEFIFirmwareDeviceArch(t *testing.T) {
+	udev := UEFIFirmwareDevice{Code: "AAVMF_CODE.ms.fd", Vars: "AAVMF_VARS.ms.fd"}
+	expected := "-drive if=pflash,format=raw,readonly=on,file=AAVMF_CODE.ms.fd -drive if=pflash,format=raw,file=AAVMF_VARS.ms.fd"
+	testAppend(udev, expected, t)
+}
+
+func TestAppendUEFIFirmwareDeviceConfigArch(t *testing.T) {
+	c := &Config{}
+	udev := UEFIFirmwareDevice{Code: "AAVMF_CODE.ms.fd", Vars: "AAVMF_VARS.ms.fd"}
+	c.UEFIFirmwareDevices = append(c.UEFIFirmwareDevices, udev)
+	err := c.appendDevices()
+	if err != nil {
+		t.Fatalf("Failed to append UEFI firwmware device: %s", err)
+	}
+	if len(c.qemuParams) == 0 {
+		t.Errorf("Expected non-empty qemuParams, found %s", c.qemuParams)
+	}
+
+	expected := "-drive if=pflash,format=raw,readonly=on,file=AAVMF_CODE.ms.fd -drive if=pflash,format=raw,file=AAVMF_VARS.ms.fd"
+	result := strings.Join(c.qemuParams, " ")
+	if expected != result {
+		t.Fatalf("Failed to append parameters\nexpected[%s]\n!=\nfound   [%s]", expected, result)
+	}
+}
+
+func TestNewFirmwareDev(t *testing.T) {
+	udev_ptr, err := NewSystemUEFIFirmwareDevice(true)
```

Comment:
```suggestion
func TestNewFirmwareDev(t *testing.T) {
        secureBoot := true
	udev_ptr, err := NewSystemUEFIFirmwareDevice(secureBoot)
```

I miss python's named values, like NewSystemUEFIFirmwareDevice(secureBoot=true), this was a nice way to self document what the parameter was doing.  But in go, best we have is naming the variable something helpful.